### PR TITLE
Refactor/pupil/organisms sub folders

### PR DESCRIPTION
**Pull request by kimon-satan**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

- I reorganised the components/organisms/pupils as we were finding it a bit unweildly
- At the same time I wrote some convenience scripts to resolve index.ts files and also remove any sneaky relative imports (which I hate)
- This has obviously effected a hell of a lot of files but in mostly superficial ways
- I checked with a local deploy in OWA and there are no breaking changes so a minor version is fine
- I would like to run the two scripts as part of the local husky hooks as they naturally enforce naming conventions and ban relative path exports but I will hold off for now 

## Testing instructions

- npm run publish:local
- use it in owa and run the test suite (NB. you have to install cloudinary because we still haven't fixed that bug)

## ACs

[x] - files better organised
[x] - all relative imports removed
[x] - storybook and test suites run
[x] - non breaking changes

### CODE REVIEW

File: src/components/organisms/teacher/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export * from "./OakSearchFilterCheckBox";
 export * from "./OakTertiaryOLNav";
+export * from "./OakUnitListItem";
```
Comment (mantagen) [inline src/components/organisms/teacher/index.ts:3]: Is there a lint rule for import order? Wondering why this wasn't already in the right place

**Review by mantagen — APPROVED** [review]
Nice PR that!

File: src/components/organisms/teacher/index.ts (hunk 1 above)
Comment (kimon-satan) [reply src/components/organisms/teacher/index.ts:3]: my script auto generates the exports for the index.ts files so overwrites the old versions

Comment (mantagen) [reply src/components/organisms/teacher/index.ts:3]: Ahh yes it's export I see now!

Comment (kimon-satan) [reply src/components/organisms/teacher/index.ts:3]: If I put it in the ci/cd then I will make it ordered to prevent confusing changes

**Review by benprotheroe — APPROVED** [review]
works locally in owa!

**Pull request merged**